src/test/java/com/teragrep/pth10/logicalOperationTest.java

```diff
@@ -260,229 +324,314 @@ public void parseWithQuotesInsideQuotesCatalystTest() {
-        result = utils.getQueryAnalysis(q);
-        assertEquals(e, result);
+            assertEquals(2, res.count());
+            assertEquals("\"raw 04\"", listOfRaw.get(0));
+            assertEquals("\"raw 05\"", listOfRaw.get(1));
+        });
     }
 
-    @Disabled
-    @Test // disabled on 2022-05-16 TODO Convert to dataframe test
-    public void streamListWithoutQuotesTest() throws AnalysisException {
-        String q, e, result;
-        q = "index = memory-test latest=\"05/10/2022:09:11:40\" host= sc-99-99-14-25 sourcetype= log:f17:0 Latitude";
-        long latestEpoch = new DefaultTimeFormat().getEpoch("05/10/2022:09:11:40");
-        e = "SELECT * FROM `temporaryDPLView` WHERE index LIKE \"memory-test\" AND _time <= from_unixtime("
-                + latestEpoch
-                + ") AND host LIKE \"sc-99-99-14-25\" AND sourcetype LIKE \"log:f17:0\" AND _raw LIKE '%Latitude%'";
-        result = utils.getQueryAnalysis(q);
-        assertEquals(e, result);
-    }
+    @Test
+    @Disabled(
+            value = "search does not allow having sourcetype in this format: sourcetype IN (sourcetyp1, sourcetype2), pth-10 issue #282"
+    )
+    public void testWithMultipleSourcetypes() {
+        String query = "\"index=index_* sourcetype IN (B:X:0, B:Y:0, C:X:0)\"";
+        this.streamingTestUtil.performDPLTest(query, this.testFile, res -> {
+            List<String> listOfSourcetype = res
+                    .select("_raw")
+                    .orderBy("offset")
+                    .collectAsList()
+                    .stream()
+                    .map(r -> r.getAs(0).toString())
+                    .collect(Collectors.toList());
 
-    @Disabled
-    @Test // disabled on 2022-05-16 TODO Convert to dataframe test
-    public void streamListWithoutQuotes1Test() throws AnalysisException {
-        String q, e, result;
-        // Test to_lower() for inex,host,sourcetyype
-        q = "index = MEMORY-test latest=\"05/10/2022:09:11:40\" host= SC-99-99-14-20 sourcetype= LOG:F17:0 Latitude";
-        long latestEpoch2 = new DefaultTimeFormat().getEpoch("05/10/2022:09:11:40");
-        e = "SELECT * FROM `temporaryDPLView` WHERE index LIKE \"memory-test\" AND _time <= from_unixtime("
-                + latestEpoch2
-                + ") AND host LIKE \"sc-99-99-14-20\" AND sourcetype LIKE \"log:f17:0\" AND _raw LIKE '%Latitude%'";
-        result = utils.getQueryAnalysis(q);
-        assertEquals(e, result);
-    }
+            assertEquals(5, res.count());
+            assertEquals("B:X:0", listOfSourcetype.get(0));
+            assertEquals("B:X:0", listOfSourcetype.get(1));
+            assertEquals("B:Y:0", listOfSourcetype.get(2));
+            assertEquals("B:Y:0", listOfSourcetype.get(3));
+            assertEquals("C:X:0", listOfSourcetype.get(4));
+        });
 
-    @Disabled
-    @Test // disabled on 2022-05-16 TODO Convert to dataframe test
-    public void logicalNotTest() throws AnalysisException {
```

Comment:
is this case with NOT tested? I can't seem to find it in the new version of this file

Reply:
never mind, found it closer to the end of the file.